test/specs/style_manager/index.ts

```diff
@@ -195,6 +195,44 @@ describe('StyleManager', () => {
         expect(obj.getSelectedParents()).toEqual([rule1]);
       });
 
+      test('With multiple classes, should both be in parents list', () => {
+        const cmp = domc.addComponent('<div class="cls cls2"></div>');
+        const [rule1, rule2] = cssc.addRules(`
+          .cls { color: red; }
+          .cls2 { color: blue; }
+        `);
+        em.setSelected(cmp);
+        obj.__upSel();
+        expect(obj.getSelected()).not.toBe(rule1);
+        expect(obj.getSelected()).not.toBe(rule2);
+        expect(obj.getSelectedParents()).toEqual([rule2, rule1]);
+      });
+
+      test('With tagName + class, class first', () => {
+        const cmp = domc.addComponent('<div class="cls" id="id-test"></div>');
+        const [rule1, rule2] = cssc.addRules(`
+          .cls { color: red; }
+          div { color: yellow; }
+        `);
+        em.setSelected(cmp);
+        obj.__upSel();
+        expect(obj.getSelected()).toBe(rule1);
+        expect(obj.getSelectedParents()).toEqual([rule2]);
+      });
+
+      test('With tagName + ID + class, class first, ID second', () => {
+        const cmp = domc.addComponent('<div class="cls" id="id-test"></div>');
+        const [rule1, rule2, rule3] = cssc.addRules(`
+          .cls { color: red; }
+          div { color: yellow; }
+          #id-test { color: blue; }
+        `);
+        em.setSelected(cmp);
+        obj.__upSel();
+        expect(obj.getSelected()).toBe(rule1);
+        expect(obj.getSelectedParents()).toEqual([rule3, rule2]);
+      });
```

Comment:
Thanks 🙇‍♂️ it's good to cover with tests this kind of logic

Reply:
It's always better when it's tested 😉